Python composite module execution unit tests on delegation of backend_with_compiler_demo

Amyh11325 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60801**

backend_with_compiler_demo

Summary:
Added unit tests for the execution of a simple composite module with a
compiler

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v gives a
success

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29409958](https://our.internmc.facebook.com/intern/diff/D29409958)

Commit:
Python composite module execution unit tests on delegation of
backend_with_compiler_demo

Summary:
Added unit tests for the execution of a simple composite module with a
compiler

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v gives a
success

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29409958).

iseeyuan commented on test/jit/test_backends.py:
```diff
@@ -558,6 +558,66 @@ def test_execution(self):
+        self.scripted_module = torch.jit.script(CompModuleTestWithCompiler.CompModule(lowered_add, lowered_sub))
+        # No backend and mobile versions of CompModule currently, so this is filler.
+        self.lowered_module = self.scripted_module
+        self.mobile_module = self.scripted_module
```
Would mobile_module work when converting from scripted_module, using _save_to_buffer_for_lite_interpreter(self.scripted_module)?

Amyh11325 replied:
It seems to work.

iseeyuan reviewed (commented):
Please check the inline comment, and see if it's possible to test mobile module from scripted.

Commit:
Update on "Python composite module execution unit tests on delegation of backend_with_compiler_demo"


backend_with_compiler_demo

Summary:
Added unit tests for the execution of a simple composite module with a
compiler

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v gives a
success

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29409958](https://our.internmc.facebook.com/intern/diff/D29409958)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29409958).

iseeyuan reviewed (approved):
LGTM. Thanks!
